### DISCUSSION TITLE
Werkzeugkasten Container "Darstellung"

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#109] [Ideas]
Da in dem Container "Darstellung" des Werkzeugkastens nur noch ein Tool übrig geblieben ist "Darstellungsfilter", könnte man diesen entweder zu den "Abfragen" oder  zur "Karte" verschieben.  

Abstimmung beim nexten webGIS-WS?

Damit wär mehr Platz für die Contiainer der CustomTools.

**Comment by gis-eni** [comment]
Es gibt auch noch den "Themen Beschriften" Button, der in diesem Container drin ist..

**Comment by gis-eni** [comment]
Idee: Prüfen, ob eine Zuordnung der Werkzeuge auf die Container in der ``custom.js`` erfolgen könnte.

**Comment by gis-eni** [comment]
Tipp: ``Deckkraft`` in ``Transparenz`` ändern

**Comment by gis-eni** [comment]
Ab Build 5.23.4501:

über die `custom.js` lassen sich Eigenschaften der Werkezuge überschreiben. Dazu braucht man die `id` des Werkzeuges. Diese findet man beispielsweise über die WebGIS API : `https://...../rest/tools`

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/d4c52a7c-04cd-46a8-9f8a-081b43f8d191)

Über die `id` kann man dann einige Eigenschaften des Werkzeuges überschreiben(``name``, ``container``,``tooltip``, ``image``, ``cursor``, ``help_urlpath``):

``` javascript
webgis.usability.toolProperties['webgis.tools.presentation.labeling'] = {
    container: 'Abfragen',
    name: 'Labelling', 
    tooltip: 'Lass dich überraschen'
};
webgis.usability.toolProperties['webgis.tools.presentation.visfilterremoveall'] = { container: 'Abfragen' };
webgis.usability.toolProperties['webgis.tools.presentation.visfilter'] = { container: 'Abfragen' };
```

Hinweis:

Möchte man so den ganzen ``Darstellung`` Container weg bringen, sollten man alle Werkzeuge in einen anderen Container schieben. Es gibt auch "versteckte" Werkzeuge wie ``webgis.tools.presentation.visfilterremoveall``, die erst sichtbar werden, wenn ein Filter ausgewählt wird. Natürlich "blockieren" die dann den Container.

**Comment by marko-jer** [comment]
Für die Custom-Tools wünsche ich mir den Parameter "help_urlpath", damit man die Anleitung für das  Custom-Tool hinter den Fragezeichnen ![image](https://github.com/gis-eni/webgis-forum/assets/1842500/18e6f364-9dbd-4021-97a5-3efa28b5186a) legen könnte , wie bei anderen Tools auch. Jetzt hat das Fragezeichen bei Custom-Tools keine Funktion!

Momentan behelfen wir uns mit den einbauen des Links in die "description".
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/66c0f0ed-0aaf-4511-924f-89b571ce7933)

**Reply by gis-eni** [reply]
Bei den Custom Tools kann ab jetzt eine Parameter `help_urlpath` angegeben werden. Darin sollte in absoluter Pfad zur einer Online Hilfe für dieses Werkzeug angegeben werden:

``` javascript
webgis.custom.tools.add({
    id: 'supertool',
    name: 'Super Tool',
    tooltype: 'click',
    command:'https://localhost:44341?wkt={wkt}',
    command_target: 'dialog',
    container: 'Metadata',
    cursor: 'crosshair',
    image: 'cursor-plus-26-b.png',
    help_urlpath: 'https://docs.webgiscloud.com/cloud/selfhosting/install/index.html'
});
```

Die Hilfe für dieses Werkzeug wird danach Hilfe Fenster angezeigt.

Wir für ein Werkzeug keine Hilfe angeboten, erscheint der Hilfebutton auch nicht mehr über dem Werkzeugdialog.
